LocalFolder/explainer.md

```diff
@@ -0,0 +1,129 @@
+
+In this explainer, we propose a solution that allows Microsoft Store PWAs to read and delete files from the `LocalFolder` directory belonging to their PFN. This allows apps to provide a more seamless user experience after an update as well as reclaim storage space.
+
+The proposed solution works by making use of the [Origin Private File System (OPFS)](https://web.dev/articles/origin-private-file-system) and the [File System Access API](https://wicg.github.io/file-system-access/). It exposes an app's LocalFolder file system directory as an entry in the app origin's OPFS root directory.
+
+## Goals
+
+* Enable a Microsoft Store PWA to:
+   * read files from the `LocalFolder` directory specific to its PFN.
+   * delete any file and sub-directory in its `LocalFolder` directory. 
+* Avoid prompting end users to provide input or confirmation.
+* Avoid naming collisions with other OPFS root entries. 
+
+## Non-goals
+
+* Enable a Microsoft Store PWA to create new files or modify existing files in its LocalFolder directory.
+* Support web apps not installed and managed by the Microsoft Store.
+* Support web apps on non-Windows platforms.
+* Standardization as a web specification.
+* Introduce a non-standard Web API
+
+## Proposed Solution 
+
+### Existing behavior
+
+[`navigator.storage.getDirectory()`](https://developer.mozilla.org/en-US/docs/Web/API/StorageManager/getDirectory) is an existing API that returns a promise to a [`FileSystemDirectoryHandle`] (https://developer.mozilla.org/en-US/docs/Web/API/FileSystemDirectoryHandle) representing the root of the origin's OPFS storage space. By default, an origin's OPFS root directory will have no entries.
+
+### New behavior
+
+Within a Microsoft Store PWA context, the directory handle for OPFS root will contain an additional entry that represents the app's LocalFolder location. This entry can be retrieved as a [`FileSystemDirectoryHandle`](https://developer.mozilla.org/en-US/docs/Web/API/FileSystemDirectoryHandle) by calling [`.entries()`](https://developer.mozilla.org/en-US/docs/Web/API/FileSystemDirectoryHandle/entries) or [`.getDirectoryHandle(...)`](https://developer.mozilla.org/en-US/docs/Web/API/FileSystemDirectoryHandle/getDirectoryHandle) on the OPFS root directory handle.
+
+
+The LocalFolder directory handle and its contents will have properties different from other OPFS directory handles - they will only have read and delete permissions instead of read and write permissions.
```

Comment:
Nit: There's either `"read"` or `"readwrite"`. There's no special delete-only case.

Reply:
I'll rephrase. The handle and its contents will have `read` permission without the ability to elevate to `readwrite` permission. The exception to this is when [`remove`](https://developer.mozilla.org/en-US/docs/Web/API/FileSystemHandle/remove) or [`removeEntry`](https://developer.mozilla.org/en-US/docs/Web/API/FileSystemDirectoryHandle/removeEntry) is used. With respect to those methods, the handles will have `readwrite` permission.